File: dotnet/cafdemalihapa/CommandLineArgs.cs
namespace cafdemalihapa
{
    public class CommandLineArgs
    {
        public static string GetByKey(string commandLineArgKey)
        {
            var commandLineArgs = Environment.GetCommandLineArgs();

            var index = Array.FindIndex(commandLineArgs, x => x.StartsWith(commandLineArgKey));
            if (index == -1) return $"""couldn't find environment variable "{commandLineArgKey}" ...""";
            var commandLineArgValue = commandLineArgs[index + 1];
            return commandLineArgValue;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/CommandLineArgs.fs
namespace cafdemwimu.console

open System

type CommandLineArgs() =
    static member GetByKey(commandLineArgKey: string) =
        let commandLineArgs = Environment.GetCommandLineArgs()

        let index = Array.FindIndex(commandLineArgs, fun x -> x.StartsWith(commandLineArgKey))
        if index = -1 then
            $"couldn't find environment variable \"{commandLineArgKey}\" ..."
        else
            commandLineArgs.[index + 1]
